ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewGroup.java

```diff
@@ -455,6 +456,50 @@ private void updateSubviewClipStatus(View subview) {
     }
   }
 
+  @Override
+  public boolean getChildVisibleRect(View child, Rect r, android.graphics.Point offset) {
+    // This is based on the Android ViewGroup implementation, modified to clip child rects
+    // if overflow is set to ViewProps.HIDDEN. This effectively solves Issue #23870 which
+    // appears to have been introduced by FLAG_CLIP_CHILDREN being forced false
+    // regardless of whether clipping is desired.
+    final RectF rect = new RectF();
+    rect.set(r);
+
+    child.getMatrix().mapRect(rect);
+
+    final int dx = child.getLeft() - getScrollX();
+    int dy = child.getTop() - getScrollY();
+
+    rect.offset(dx, dy);
+
+    if (offset != null) {
+      float[] position = new float[2];
+      position[0] = offset.x;
+      position[1] = offset.y;
+      child.getMatrix().mapPoints(position);
+      offset.x = Math.round(position[0]) + dx;
+      offset.y = Math.round(position[1]) + dy;
+    }
+
+    final int width = getRight() - getLeft();
+    final int height = getBottom() - getTop();
+
+    boolean rectIsVisible = true;
+
+    ViewParent parent = getParent();
+    if (parent == null || ViewProps.HIDDEN.equals(mOverflow)) {
+      rectIsVisible = rect.intersect(0, 0, width, height);
+    }
```

Comment:
What's about the clip-to-padding flag (i.e. `FLAG_CLIP_TO_PADDING`) - should you not keep this test as well?
ref: https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/view/ViewGroup.java#6229

Reply:
@d4vidi, yeah I considered that too, but again due to access level it and also maintaining Android API 16 compatibility, it didn't appear to be possible.